illumos-utils/src/zfs.rs

```diff
@@ -23,13 +21,21 @@ pub struct ListDatasetsError {
     err: crate::ExecutionError,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum DestroyDatasetErrorVariant {
+    #[error("Dataset not found")]
+    NotFound,
+    #[error(transparent)]
+    Other(crate::ExecutionError),
+}
+
```

Comment:
We have a more specific error here so that the caller can get a better idea of if we tried to delete a dataset that doesn't exist (I mean, go figure, but it's relevant now because we're doing this on boot for the zone datasets).